qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/UserAccountSteps.java

```diff
@@ -139,11 +139,65 @@ public void enterDoctorDetails() {
 	@And("a user clicks on save account button")
 	public void clickOnSaveUserAccountButton() {
 		userAccountPage.saveUserAccount();
-		userAccountPage.waitForPage();
 	}
 
 	@Then("the system adds user account into the users table")
 	public void systemAddsUserAccount() {
 		assertTrue(textExists("Account Saved Successfully"));
 	}
+
+	@Then("the system confirms password meets the password criteria")
+	public void systemConfirmsPasswordMeetsPasswordCriteria() {
+		assertFalse(textExists("Validation errors found Failed to save account details"));
+	}
+
+	@And("a user enters person details in the user account form") 
+	public void enterPersonalDetails() {
+		userAccountPage.enterPersonalDetails("Clerk", "Data");
+		userAccountPage.selectGender();
+	}
+
+	@And("a user enters user account details in the user account form")
+	public void enterUserAccountDetails() {
+		userAccountPage.clickOnAddUserAccount();
+		userAccountPage.setUsername(CLERK);
+		userAccountPage.setUserPrivilegeLevel("Full");
+	}
+
+	@And("a user sets the password with only upper case letters")
+	public void setPasswordWithOnlyUpperCaseLetters() {
+		userAccountPage.setUserPassword("DATACLERK123", "DATACLERK123");
+	}
+
+	@And("a user sets the password with only digits")
+	public void setPasswordsWithOnlyDigits() {
+		userAccountPage.setUserPassword("123459876", "123459876");
+	}
+
+	@And("a user sets the password with only letters")
+	public void setPasswordsWithOnlyLetters() {
+		userAccountPage.setUserPassword("dataclerk", "dataclerk");
+	}
+
+	@And("a user sets the password which is lower than 8 characters")
+	public void setPasswordBelowTheMinimumLength() {
+		userAccountPage.setUserPassword("123Abc", "123Abc");
+	}
+
+	@And("a user enters password that meets the password criteria")
+        public void userEnterPasswordThatMeetsThePasswordCriteria() {
+                userAccountPage.setUserPassword("Dataclerk123", "Dataclerk123");
+        }
+
+	@Then("the system throws validation error on the password input field")
+	public void systemThrowsValidationErrorOnPasswordInputField() {
+		assertFalse(userAccountPage.containsText(SYSTEM_ALERT));
+	}
+
+	@Then("the system throws a validation error message")
+        public void systemThrowsAvalidationError() {
+			List<String> validationErrors = userAccountPage.getValidationErrors();
+            userAccountPage.isValidationErrorsFound(validationErrors);
```

Comment:
We need to ensure the test validates that the exact error is being thrown when entered data is not valid.  I think we can use something like;
```js
assertTrue(userAccountPage.getValidationErrors().contains("error message"));
```